backend/Tim.Backend/Models/KustoQuery/KustoQueryRun.cs

```diff
@@ -5,31 +5,48 @@
 namespace Tim.Backend.Models.KustoQuery
 {
     using System;
-    using Tim.Backend.DataProviders.Clients;
+    using System.Collections.Generic;
 
     /// <summary>
-    /// This object represents an instance of an executed query
-    /// Query can be a saved one or simply a temporary one that was passed over via a request
-    /// In tha same the query is temporary, the Query GUid will be an empty Guid
-    /// otherwise the query guid will represent the Guid that can be used to find the actual query in the SavedQueries table in cosmosdb.
+    /// Query Run states.
     /// </summary>
-    public class KustoQueryRun
+    public enum QueryRunStates
+    {
+        /// <summary>
+        /// Query first created.
+        /// </summary>
+        Created,
+
+        /// <summary>
+        /// Query completed.
+        /// </summary>
+        Completed,
+
+        /// <summary>
+        /// Query had an error.
+        /// </summary>
+        Error,
+
+        /// <summary>
+        /// Query timed out.
+        /// </summary>
+        TimedOut,
+    }
+
+    /// <summary>
+    /// This object represents an instance of an executed query.
+    /// </summary>
+    public class KustoQueryRun : IJsonEntity
     {
         /// <summary>
         /// Initializes a new instance of the <see cref="KustoQueryRun"/> class.
         /// </summary>
-        /// <param name="triggeredBy">Who the query run was triggered by.</param>
-        /// <param name="queryId">Query to be run.</param>
-        /// <param name="start">Start time window of query to be executed.</param>
-        /// <param name="end">End time window of query to be executed.</param>
-        public KustoQueryRun(string triggeredBy, Guid queryId, DateTime start, DateTime end)
+        /// <param name="query">Query details.</param>
+        public KustoQueryRun(KustoQuery query)
         {
-            TriggeredBy = triggeredBy;
-            QueryId = queryId;
+            KustoQuery = query;
             QueryRunId = Guid.NewGuid();
-            QueryStartDateTimeUtc = start;
-            QueryEndDateTimeUtc = end;
-            TriggeredDateTimeUtc = DateTime.UtcNow;
+            ExecuteDateTimeUtc = DateTime.UtcNow;
```

Comment:
you dont want to track when the request to execute was received and when it completed? why not? would be helpful when debugging query errors in the future

Reply:
or is the assumption query stats handles that better?

Reply:
QueryStart and QueryEnd are for the time windowing for a query. This is already available in the KustoQuery object.